[Hong Ray] iP

HongRay opened the pull request:
JaynePro
JaynePro frees your mind of having to remember things you need to do, with a little sassiness. It's 

- text-based
- easy to learn
- Fast

All you need to do is,

1. Download it from here [https://github.com/HongRay/ip]
2. Clone the Repro
3. Add your tasks
4. Let it manage for you 😄 

And it is **FREE!!!**

Features:

- [x] Managing task
- [x] Managing deadline 

> [!NOTE]
> It is still not full proof though hahahah 

> If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

HongRay committed:
Testing

HongRay committed:
Testing

HongRay committed:
Level-0: Rename, Greet, Exit

HongRay committed:
Level-1: Echo

HongRay committed:
Level-0

HongRay committed:
Level-1

HongRay committed:
Level-2

HongRay committed:
Level-2

HongRay committed:
Level-3

HongRay committed:
Level-4

HongRay committed:
A-TextUiTesting

HongRay committed:
Level-5

HongRay committed:
Level-6

HongRay committed:
Level-7

HongRay committed:
Add test files

HongRay committed:
Level-8

HongRay committed:
Level-8

HongRay committed:
Merge branch 'branch-Level-8'

HongRay committed:
A-MoreOOP

HongRay committed:
Merge branch 'A-MoreOOP'

HongRay committed:
A-Packages

HongRay committed:
Merge branch 'A-Packages'

HongRay committed:
Merge branch 'add-gradle-support'

HongRay committed:
implement JUnit

HongRay committed:
Testing files

HongRay committed:
Merge branch 'branch-A-JUnit'

HongRay committed:
A-JavaDocs

HongRay committed:
A-CodingStandards

HongRay committed:
branch-Level-9

HongRay committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/jayne/Handler.java
#	src/main/java/jayne/Parser.java
#	src/main/java/jayne/task/TaskList.java

HongRay committed:
A-CodingStandard

HongRay committed:
Merge remote-tracking branch 'origin/branch-Level-9'

# Conflicts:
#	src/main/java/jayne/Handler.java
#	src/main/java/jayne/JayneException.java
#	src/main/java/jayne/Parser.java
#	src/main/java/jayne/Ui.java
#	src/main/java/jayne/task/Task.java

HongRay committed:
A-CheckStyle

HongRay committed:
A-CheckStyle

HongRay committed:
Merge branch 'branch-A-CheckStyle'

HongRay committed:
Implement GUI

Successfully implemented GUI

Problems:
Unable to integrate Jayne logic into GUI

HongRay committed:
Implement Jayne Logic

Successfully some of Jayne Android Logic

Problems:
Unable to exit GUI after Bye
Unable to show display message upon start

Future fixes:
1. Improve design

HongRay committed:
Solved Jayne Logic problems

Successfully ALl of Jayne Android Logic

Future fixes:
1. Improve design

HongRay committed:
Improve design

Feature:
- Added colour
- Added shape to image
- added margin and spacing

HongRay committed:
Improve Checkstyle

HongRay committed:
Remove duke

HongRay committed:
Merge branch 'branch-GUI'

HongRay committed:
Attempted to implement Varargs

Remove all unnecessary fields and methods.

HongRay committed:
Merge branch 'branch-A-Varargs'

jyztintan commented on src/main/java/jayne/Jayne.java:
```diff
@@ -0,0 +1,52 @@
+ * such as Ui, TaskList, and Storage, and contains the main program loop.
+ */
+public class Jayne {
+    private boolean isEnd = true;
+    private TaskList taskList;
+    private Storage storage;
```
Appropriate variable naming, abiding by camelCase and understandable variable names.

jyztintan commented on src/main/java/jayne/JayneException.java:
```diff
@@ -0,0 +1,140 @@
+ * This class extends the standard Exception class and provides specific error messages
+ * for various error scenarios encountered in the application.
+ */
+public class JayneException extends Exception {
```
Great that you followed the PascalCase convention for class names!

jyztintan commented on src/main/java/jayne/MainWindow.java:
```diff
@@ -0,0 +1,73 @@
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
```
I liked that you followed the appropriate method naming convention, using verbs and written in camelCase

jyztintan commented on src/main/java/jayne/MainWindow.java:
```diff
@@ -0,0 +1,73 @@
+    @FXML
+    private TextField userInput;
+    private Jayne jayne;
+    private boolean isEnd = false;
```
boolean variable follows naming convention of incorporating "is/has/was" 👍

jyztintan reviewed (commented):
Great job on adherring to coding standards :)

HongRay committed:
Update dialog box

Dialog box's height was not infinity, causing a truncated text when list contains more than a certain number of items.

In DialogBox.fxml, under fx:id=dialog, I added minHeight = Infinity.

This solved the problem of a truncated text. Solution and bug found was thanks to Xiao Yang from my team. Props to him.

HongRay committed:
DialogText box: make list not truncated

DialogText box truncates text.

A dialog text box that allows not truncated text will allow user to see their entire list of task.

Lets,
- Update the text box to allow infinite height
- Change Jayne reply to snowiee when bye command issued

irwynliong commented on src/main/java/jayne/Handler.java:
```diff
@@ -0,0 +1,185 @@
+ * and invoking the appropriate actions on the TaskList.
+ * It acts as a bridge between the user interface and the data model.
+ */
+public class Handler {
```
Good use of a Handler class to execute the different commands.

irwynliong reviewed (commented):
Overall good coding practice!

HongRay committed:
Generate Jar file

Generated Jayne-android jar file.

Manage to generate a Jayne-android file, however, the jar file is 11kb and still requires path from resources/Images to operate. Have not tried running jar file as standalone file yet.

Future problems,
- Make Jar file contain all dependencies and libraries and do not need to rely on other paths.

HongRay committed:
Add assertions

Add assertions to certain parts of the codes.

There are still places for improvements but this will do for now.

HongRay committed:
Merge pull request #1 from HongRay/branch-A-Assertions

Add assertions

HongRay committed:
Improves Code Quality

Improvements include:
1. Replace magic literals
2. Avoid long methods
3. Avoid deep nesting
4. Avoid complicated expressions
5. KISSing
6. SLAP
7. Make happy path prominent

Bug fix for findTask where output is not appended and thus only displayed 1 task found with keyword.

HongRay committed:
Merge pull request #2 from HongRay/branch-A-CodeQuality

Improve Code Quality

HongRay committed:
Add CI

Add gradle.yml file. Not really sure what it does so hopefully it doesnt break my system

HongRay committed:
Merge branch 'branch-A-CI'

HongRay committed:
Add sort function

Jayne is now able to sort task by type, whether its marked or not and the earliest date the event/deadline is due.

HongRay committed:
Merge branch 'branch-C-Sort'

HongRay committed:
Already better GUI

Based on previous iterations, GUI is already enhanced.

No more adjustment needs to be made.

HongRay committed:
Merge branch 'branch-A-BetterGUI'

HongRay committed:
Already better personality

Based on previous iterations, personality is included in jayne android.

No more adjustment needs to be made.

HongRay committed:
Merge branch 'branch-A-Personality'

HongRay committed:
Create Docs

Updated README.md file in /docs to reflect jayne androids features instead of duke's features

Additional stuff:
- Remove redundant code in Parser class
- Added UI.png picture in /docs

HongRay committed:
Merge branch 'branch-A-UserGuide'

HongRay committed:
Move README.md file to main

Moved updated README.md file to main directory

HongRay committed:
Update README.md

corrected Ui.png

HongRay committed:
Update README.md file to main

Moved updated README.md file to main directory

HongRay committed:
Merge branch 'branch-A-Release'

HongRay committed:
Update README.md file to main

Moved updated README.md file to main directory

HongRay committed:
Add Jar in main directory

Forgot that builds is in gitignore.
Added jar file in main directory now

HongRay committed:
Update README.md

update readme

HongRay committed:
Update README.md

Updated ReadME for Jayne.jar

Gave `jayne.jar` to make it look more representable.
User can see it easily now
